crates/reporting/tests/test_reporting.rs

```diff
@@ -4837,14 +4837,27 @@ mod test_reporting {
             "#
         ),
         @r###"
-    ── MISSING EXPRESSION ───────────────────── tmp/pattern_binds_keyword/Test.roc ─
+    ── MISSING ARROW ────────────────────────── tmp/pattern_binds_keyword/Test.roc ─
```

Comment:
Hm, the new error message still seems similarly confusing to the old one. 😅 

Really the problem is that we expected a pattern, but got a reserved keyword instead. But that's out of scope for this PR!